internal/pkg/util/fs/squashfs/squashfs.go

```diff
@@ -52,3 +55,45 @@ func GetMem() (string, error) {
 
 	return mem, err
 }
+
+var (
+	setuidMountKnown   bool
+	setuidMountAllowed bool
+)
+
+// SetuidMountAllowed calculates whether or not it is allowed to
+// mount a squashfs filesystem using the kernel driver in setuid mode.
+func SetuidMountAllowed(cfg *apptainerconf.File) bool {
+	if setuidMountKnown {
+		return setuidMountAllowed
+	}
+	setuidMountKnown = true
+	str := cfg.AllowSetuidMountSquashfs
+	if !namespaces.IsUnprivileged() {
+		setuidMountAllowed = true
+		sylog.Debugf("Kernel squashfs mount allowed because running as root")
+	} else if str == "yes" {
+		setuidMountAllowed = true
+		sylog.Debugf("Kernel squashfs mount allowed by configuration")
+	} else if str == "iflimited" {
+		if len(cfg.LimitContainerOwners) > 0 ||
+			len(cfg.LimitContainerGroups) > 0 ||
+			len(cfg.LimitContainerPaths) > 0 {
+			setuidMountAllowed = true
+			sylog.Debugf("Kernel squashfs mount allowed because of limit container")
+		} else {
+			eclcfg, err := syecl.LoadConfig(buildcfg.ECL_FILE)
+			if err != nil {
+				sylog.Debugf("Kernel squashfs mount not allowed because error loading %s: %v", buildcfg.ECL_FILE, err)
```

Comment:
This should be a warning at least, ideally we should fail as it means there is something wrong with the installation

Reply:
There's already a check for that elsewhere so I don't think it needs to be here.  If I rename ecl.toml I get a fatal error
```
FATAL:   /etc/apptainer/ecl.toml must be owned by root
```